### DISCUSSION TITLE
Is this a bad practice?

### DISCUSSION THREAD

**Discussion by temcewen** [rrousselGit/riverpod#2409] [General]
For a long time, I wanted to be able to give a family modified NotifierProvider some initial properties that weren't part of its family input. 

Example: Say I have a car model base class with properties like numberOfWheels, color, year, etc. Say I want to give it a family modifier that takes a person's name to identify it as "their" car. However, the issue is that if I want the NotifierProvider to know what color the car is assuming the color is customizable, I also have to give that in the family input as well. Then, in order to access that NotifierProvider in the future, I have to request it with all of the provided properties, when I really want to just reference it by the person's name.

I realized that this could all be possible if I just made the family input a custom class where the hashcode was only generated based upon the unique properties that I wanted it to use such as the person's name. This would allow me to give the car initial properties that weren't required in the future when I wanted to access it again. 

Is that a bad practice? If so, why? Is it okay in certain circumstances?

**Comment by rrousselGit** [comment]
Definitely bad practice :P.

Don't mess with the hashCode of family parameters just to get away with what you're trying to do.
It'll be brittle, and you'll get a race condition. (the provider could be read elsewhere with different "initial values").

Family parameters aren't meant to be "initial properties". It's voluntary as this is what empowers cool reactive stuff.

You should instead have your provider determine its initial state on its own. There's most definitely a better solution

**Reply by temcewen** [reply]
That makes sense. I'll try to create a provider that can give those initial properties then.
